packages/cli-platform-ios/src/commands/runIOS/index.ts

```diff
@@ -83,8 +82,31 @@ function runIOS(_: Array<string>, ctx: Config, args: FlagsT) {
     } "${chalk.bold(xcodeProject.name)}"`,
   );
 
-  const devices = getDevices();
+  if (args.listDevices) {
+    if (args.device || args.udid) {
+      logger.warn(
+        `Both ${
+          args.device ? 'device' : 'udid'
+        } and "list-devices" parameters were passed to "run" command. We will list available devices and let you choose from one.`,
+      );
+    }
+    const availableDevices = await listIOSDevices();
+    const selectedDevice = await promptForDeviceSelection(availableDevices);
+    if (!selectedDevice) {
+      throw new CLIError(
+        'Failed to select device, please try to run app without "list-devices" command.',
+      );
+    }
+    if (selectedDevice.type === 'simulator') {
+      return runOnSimulator(xcodeProject, scheme, args);
```

Comment:
There's no `selectedDevice` passed to the `runOnSimulator` function. 
```suggestion
      return runOnSimulator(xcodeProject, scheme, args, selectedDevice);
```